c/agoo-c/simple.c

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 by Peter Ohler, All Rights Reserved
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <agoo.h>
+#include <agoo/log.h>
+#include <agoo/page.h>
+#include <agoo/server.h>
+
+static int	user_off = 6;
+
+static void
+user_handler(Req req) {
+    agoo_respond(req, 200, req->path.start + user_off, req->path.len - user_off, NULL);
+}
+
+static void
+empty_handler(Req req) {
+    agoo_respond(req, 200, NULL, 0, NULL);
+}
+
+int
+main(int argc, char **argv) {
+    struct _Err	err = ERR_INIT;
+    
+    agoo_init("simple");
+
+    // Set the number of eval threads.
+    the_server.thread_cnt = 1;
```

Comment:
Can't this be hardcoded to the number of cores on the system or so instead?

Reply:
With the light weight functions it doesn't make sense to set it to anything more than 1. If the functions were long running such as hitting a database then having more threads makes sense. You could set it to 100 if you wanted usually the developer tunes the thread count according to the application. It is not really related to the number of cores. As it is there are at least 5 threads running in Agoo-C by default when there are 1000 connections. At some point the `thread_cnt` option will be removed in favor of a self tuning thread creator. Its not there yet but if it was the count would be 1.

Long winded reply.

Reply:
Long winded reply's are often good for the information they give though.  ^.^

So it does already use a thread pool for receiving network connections regardless?  That's all good then.  :-)

Reply:
Sort of. There is a single thread for accepting connections which get handed off to the read/write threads. If the read/write threads get too busy (exceed some managed connection limit) then a new read/write thread is created. There is a maximum number of threads imposed but that, for now is hard coded.

Reply:
it's kind of https://github.com/libuv/libuv then  ?

Reply:
Not nearly as expansive but could be doing some of the same things.